coursemapper-kg/recommendation/app/services/course_materials/recommendation/recommender.py

```
@@ -206,198 +268,111 @@ def recommend(
-                flush=True,
-            )
+    def recommend(
+        self,
+        recommendation_type=RecommendationType.WITHOUT_EMBEDDING,
+        data:pd.DataFrame=None,
+        slide_weighted_avg_embedding_of_concepts="",
+        slide_document_embedding="",
+        user_embedding="",
+        top_n=10,
+        video=True,
+        not_understood_concept_list=[],
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)